src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
-        return null;
+        return searchNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode;
+        T e = null;
+        currentNode = searchNodeByIndex(index);
+        e = currentNode.item;
+        currentNode.item = value;
+        return e;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T deleteElement = null;
+        if (checkIndex(index)) {
+            deleteElement = (index == 0) ? unlinkFirst()
+                    : (index == size - 1) ? unlinkLast()
+                    : unlink(searchNodeByIndex(index));
+        }
+        return deleteElement;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = searchNodeByElement(object);
+        if (currentNode == null) {
+            return false;
+        } else if (currentNode.equals(head)) {
+            unlinkFirst();
+            return true;
+        } else if (currentNode.equals(tail)) {
+            unlinkLast();
+            return true;
+        }
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    public T unlink(Node<T> object) {
+        Node<T> currentNode = object;
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return currentNode.item;
+    }
+
+    public T unlinkFirst() {
+        T deleteElement = null;
+        if (size == 1) {
+            deleteElement = head.item;
+            tail = null;
+            head = null;
+        } else {
+            deleteElement = head.item;
+            head = head.next;
+            head.prev = null;
+        }
+        size--;
+        return deleteElement;
+    }
+
+    public T unlinkLast() {
+        T deleteElement = null;
+        deleteElement = tail.item;
+        tail = tail.prev;
+        tail.next = null;
+        size--;
+        return deleteElement;
+    }
+
+    public Node<T> searchNodeByElement(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.item, object)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    public Node<T> searchNodeByIndex(int index) {
+        if (checkIndex(index)) {
+            Node<T> currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode;
+        }
+        return null;
+    }
+
+    public boolean checkIndex(int index) {
```

Comment:
Naming should give idea of return type. checkIndex looks like void method, not boolean
```suggestion
    public boolean isValidIndex(int index) {
```

But as you throw exception inside, this method cannot return false at all. So maybe better to make it void and call without if statements in place of usage?
```suggestion
    public void checkIndex(int index) {
```